hal/src/nRF52840/adc_hal.cpp

```diff
@@ -137,9 +138,13 @@ void hal_adc_dma_init() {
     if (ret == SYSTEM_ERROR_NONE && hwVersion == HAL_VERSION_BRN404X_V003) {
         adcReference = HAL_ADC_REFERENCE_INTERNAL;
     }
-#else
-    (void)adcReference;
 #endif // PLATFORM_ID == PLATFORM_BORON
+
+    hal_device_hw_info hwInfo = {};
```

Comment:
size needs to be set, see resolved comment

Reply:
ok, is this checked anywhere? What about version? Or is this just implicitly version 0?

Reply:
It's going to be checked later on if that struct is extended and one is going to forget to update this particular instance of a call to `hal_device_hw_info()` and it will silently stop working. All sized structs need to have size initialized. Version can be kept as-is/default.

Reply:
Ok got it 👍 will keep that in mind for similar structs in the future